runtime/appruntime/apisdk/api/handler.go

```diff
@@ -691,15 +681,20 @@ func (d *Desc[Req, Resp]) externalCall(c CallContext, service config.Service, re
 				if err != nil {
 					return resp, errs.B().Code(errs.Internal).Msg("request failed: unable to read response").Err()
 				}
-				return resp, errs.B().Code(errs.Internal).Msg("request failed: " + string(bodyBytes)).Err()
+
+				if len(bodyBytes) == 0 {
+					return resp, errs.B().Code(errs.Internal).Msgf("request failed: status %d (%s)", httpResp.StatusCode, httpResp.Status).Err()
+				} else {
+					return resp, errs.B().Code(errs.Internal).Msgf("request failed: status %d: %s", httpResp.StatusCode, string(bodyBytes)).Err()
+				}
```

Comment:
Turn's out we had a shit error experience if there was no body returned, so this is just to make it easier to understand what's gong on

Reply:
Makes sense. Maybe just use `httpResp.Status` in both cases instead of `status %d`? `httpResp.Status` includes the HTTP status code but also what it is ("500 Internal Server Error")